src/Core/tests/UnitTests/Hosting/RegisteredHandlerServiceTypeSetTests.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using Microsoft.Maui.Hosting.Internal;
+using Microsoft.Maui.Platform;
+using Microsoft.Maui.UnitTests;
+using Xunit;
+
+namespace Microsoft.Maui.UnitTesting.Hosting
+{
+	[Category(TestCategory.Core, TestCategory.Hosting)]
+	public class RegisteredHandlerServiceTypeSetTests
+	{
+		[Fact]
+		public void ResolveCorrespondingHandlerServiceType()
+		{
+			var registeredTypes = new RegisteredHandlerServiceTypeSet();
+			registeredTypes.Add(typeof(ViewStub));
+
+			var type = registeredTypes.ResolveVirtualViewToRegisteredHandlerServiceType(typeof(ViewStub));
+
+			Assert.Equal(typeof(ViewStub), type);
+		}
+
+		[Fact]
+		public void ThrowsWhenNoMatchingHandlerServiceTypeIsRegistered()
+		{
+			var registeredTypes = new RegisteredHandlerServiceTypeSet();
+
+			Assert.Throws<HandlerNotFoundException>(() => registeredTypes.ResolveVirtualViewToRegisteredHandlerServiceType(typeof(ViewStub)));
+		}
+
+		[Theory]
+		[InlineData(typeof(IViewStub))]
+		[InlineData(typeof(IMyBaseViewStub))]
+		[InlineData(typeof(IMyDerivedViewStub))]
+		public void ResolvesBaseInterfaceServiceType(Type baseInterfaceType)
+		{
+			var registeredTypes = new RegisteredHandlerServiceTypeSet();
+			registeredTypes.Add(baseInterfaceType);
+
+			var resolvedServiceType = registeredTypes.ResolveVirtualViewToRegisteredHandlerServiceType(typeof(MyDerivedViewStub));
+
+			Assert.NotNull(resolvedServiceType);
```

Comment:
Should `resolvedServiceType` be equal to `baseInterfaceType`?

Reply:
We could add that check. I figured that if it was not null, and didn't throw for that matter, that there was a match since there was only one type added to the set.